fragmenter/chemi.py

```diff
@@ -1,1619 +1,525 @@
-        charged_copy.SetCoords( original )
-    elif keep_confs > 0:
-        logger().debug("keep_confs was set to %s. Molecule positions will be reset." % keep_confs)
-
-        #Otherwise if a number is provided, return this many confs if available
-        for k, conf in enumerate( charged_copy.GetConfs() ):
-            if k > keep_confs - 1:
-                charged_copy.DeleteConf(conf)
-    elif keep_confs == -1:
-        #If we want all conformations, continue
-        pass
-    else:
-        #Not a valid option to keep_confs
-        raise(ValueError('Not a valid option to keep_confs in get_charges.'))
-
-    return charged_copy
-
-
-def generate_conformers(molecule, max_confs=800, dense=False, strict_stereo=True, ewindow=15.0, rms_threshold=1.0, strict_types=True,
-                        can_order=True, copy=True, timeout=100):
-    """Generate conformations for the supplied molecule
-    Parameters
-    ----------
-    molecule : OEMol
-        Molecule for which to generate conformers
-    max_confs : int, optional, default=800
-        Max number of conformers to generate.  If None, use default OE Value.
-    strict_stereo : bool, optional, default=True
-        If False, permits smiles strings with unspecified stereochemistry.
-    strict_types : bool, optional, default=True
-        If True, requires that Omega have exact MMFF types for atoms in molecule; otherwise, allows the closest atom
-        type of the same element to be used.
     Returns
     -------
-    molcopy : OEMol
-        A multi-conformer molecule with up to max_confs conformers.
-    Notes
-    -----
-    Roughly follows
-    http://docs.eyesopen.com/toolkits/cookbook/python/modeling/am1-bcc.html
+        A molecule which contains ELF10 AM1 Wiberg Bond orders.
     """
-    if copy:
-        molcopy = oechem.OEMol(molecule)
-    else:
-        molcopy = molecule
-
-    if dense:
-        omega_opts = oeomega.OEOmegaOptions(oeomega.OEOmegaSampling_Dense)
-        omega = oeomega.OEOmega(omega_opts)
-    else:
-        omega = oeomega.OEOmega()
 
-    atom_map = False
-    if cmiles.utils.has_atom_map(molcopy):
-        atom_map = True
-        cmiles.utils.remove_atom_map(molcopy)
+    molecule = copy.deepcopy(molecule)
 
-    # These parameters were chosen to match http://docs.eyesopen.com/toolkits/cookbook/python/modeling/am1-bcc.html
-    omega.SetMaxConfs(max_confs)
-    omega.SetIncludeInput(True)
-    omega.SetCanonOrder(can_order)
+    # Store the atom map separately in case it gets removed by a TK.
+    atom_map = molecule.properties.pop("atom_map", None)
 
-    omega.SetSampleHydrogens(True)  # Word to the wise: skipping this step can lead to significantly different charges!
-    omega.SetEnergyWindow(ewindow)
-    omega.SetRMSThreshold(rms_threshold)  # Word to the wise: skipping this step can lead to significantly different charges!
+    # Generate a set of ELF10 conformers.
+    molecule = generate_conformers(molecule, max_confs)
```

Comment:
If this is the only place you call `generate_conformers`, do you need to pop the atom map and deepcopy the molecule twice?

Reply:
Because it is part of the public API we should probably leave this in for now. Whether this should be part of the public API is a different matter however!